src/components/periodic-report/periodic-report.service.ts

```diff
@@ -152,13 +157,68 @@ export class PeriodicReportService {
     };
   }
 
+  async getSingleProjectReport(
+    projectId: ID,
+    reportType: ReportType,
+    startDate: DateTime,
+    session: Session
+  ): Promise<PeriodicReport | null> {
+    const result = await this.repo.currentProjectReport(
+      projectId,
+      reportType,
+      startDate
+    );
+
+    if (!result) {
+      return null;
+    }
+
+    return await this.readOne(result.id, session);
+  }
+
+  async getDueProjectReport(
+    projectId: ID,
+    reportType: ReportType,
+    session: Session
+  ): Promise<DueReport> {
+    let interval: DurationUnit = 'quarter';
+    if (reportType === ReportType.Financial) {
+      const project = await this.project.readOne(projectId, session);
+      if (project.financialReportPeriod.value === ReportPeriod.Monthly) {
+        interval = 'month';
+      }
+    }
+
+    const currentReportStartDate = DateTime.now()
```

Comment:
```suggestion
    const currentReportStartDate = CalendarDate.local()
```

I think DateTime.now() is new, if you want to use CalendarDate.now() you'll need to add it to our class.